galaxy_ng/app/api/v1/viewsets/tasks.py

```diff
@@ -0,0 +1,114 @@
+import datetime
+import logging
+
+from django.conf import settings
+
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from pulpcore.plugin.models import Task
+from pulpcore.plugin.tasking import dispatch
+
+from galaxy_ng.app.access_control.access_policy import LegacyAccessPolicy
+
+from rest_framework.settings import perform_import
+
+
+GALAXY_AUTHENTICATION_CLASSES = perform_import(
+    settings.GALAXY_AUTHENTICATION_CLASSES,
+    'GALAXY_AUTHENTICATION_CLASSES'
+)
+
+logger = logging.getLogger(__name__)
+
+
+class LegacyTasksViewset:
+    """
+    Legacy task helper.
+
+    v1 task ids were integer values and have a different
+    schema and states from the pulp tasking system.
+
+    This set of functions will reshape a pulp task into
+    something compatible with v1 clients such as the
+    galaxy cli.
+    """
+
+    permission_classes = [LegacyAccessPolicy]
+    authentication_classes = GALAXY_AUTHENTICATION_CLASSES
+
+    def legacy_task_hash(self, pulp_id):
+        """Transform a uuid into an integer."""
+        return abs(hash(str(pulp_id)))
+
+    def legacy_dispatch(self, function, kwargs=None):
+        """Dispatch wrapper for legacy tasks."""
+        task = dispatch(function, kwargs=kwargs)
+        hashed = self.legacy_task_hash(task.pulp_id)
+        return hashed
+
+    @action(detail=True, methods=['get'], name="Get task")
+    def get_task(self, request, id=None):
+        """Get a pulp task via the transformed v1 integer task id."""
+        logger.debug(f'GET TASK: {request}')
+        if id:
+            task_id = id
+        else:
+            task_id = int(request.GET.get('id', None))
+        logger.debug(f'GET TASK id: {task_id}')
+
+        # iterate through most recent tasks to find the matching uuid
+        this_task = None
+        for t in Task.objects.all().order_by('started_at').reverse():
```

Comment:
This system is going to have tens of thousands of tasks in it. There has to be a more efficient way to find the task than just iterating over each one. This will make it super easy to DoS the system by simply spamming api requests to this api endpoint with an task ID that doesn't exist